src/arduino.cc/arduino-builder/main.go

```diff
@@ -249,8 +248,7 @@ func main() {
 		_, err := os.Stat(buildPath)
 		if err != nil {
 			fmt.Fprintln(os.Stderr, err)
-			defer os.Exit(1)
-			return
+			os.Exit(1)
```

Comment:
Oh good catch, that's a nice one that I forget to comment on the PR.
I kept this change in a single commit so it can be easily reverted in case.

> IIRC deferring the exit was useful to make sure other defer and error handlers run

what you said applies for deferred actions created in the current scope (i.e. the current running function and its callers), but since the `main` function, basically, does only command line parsing all the defers are instantiated inside "Commands" and when the control returns back to the main function all the internal defers are already out of scope and executed. Since the `main` function doesn't use `defer` I opted to skip the `defer os.Exit` construct.

The drawback is that new defers may be added in the `main` function in the future and it would be more future-proof to keep the `defer os.Exit` construct instead. Personally I prefer the "classic" behavior of the "exit" function, where the execution of the program is terminated immediately.